src/tabbar.cpp

```diff
@@ -49,20 +54,25 @@ TabBar::TabBar(QWidget *parent) :
             });
     connect(app->getAction(KiwixApp::SettingAction), &QAction::triggered,
             this, [=]() {
-                if (KiwixApp::instance()->getSettingsManager()->isSettingsViewdisplayed()) {
-                    setCurrentIndex(m_settingsIndex);
-                    return;
+                for (int i = 0 ; i < (mp_stackedWidget->count() - 1) ; i++) {
+                    if (qobject_cast<SettingsManagerView*>(mp_stackedWidget->widget(i))) {
+                        setCurrentIndex(i);
+                        return;
+                    }
                 }
-                auto index = currentIndex() + 1;
-                m_settingsIndex = index;
-                auto view = KiwixApp::instance()->getSettingsManager()->getView();
+                int index = currentIndex() + 1;
+                SettingsManagerView* view = KiwixApp::instance()->getSettingsManager()->getView();
```

Comment:
Because `auto` decreases code readability here. I have to know which type is returned by `getView()`

https://wiki.qt.io/Coding_Conventions

> auto Keyword
> Optionally, you can use the auto keyword in the following cases. If in doubt, for example if using auto could make the code less readable, do not use auto. Keep in mind that code is read much more often than written. 

There also examples when to use `auto`. This piece of code doesn't fall into it.